File: Set1_1000/2_Add_Two_Numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        if (l1 == null && l2 == null)
            return null;
        ListNode dummy = new ListNode(0);
        ListNode l3 = dummy;
        int carry = 0;
        while (l1 != null || l2 != null) {
            int val1 = (l1 != null) ? l1.val : 0;
            int val2 = (l2 != null) ? l2.val : 0;
            int sum = val1 + val2 + carry;
            carry = sum / 10;
            int lastDigit = sum % 10;
            ListNode currentLastDigit = new ListNode(lastDigit);
            l3.next = currentLastDigit;
            if (l1 != null)
                l1 = l1.next;
            if (l2 != null)
                l2 = l2.next;
            l3 = l3.next;
        }
        if (carry > 0) {
            ListNode currentLastDigit = new ListNode(carry);
            l3.next = currentLastDigit;
            l3 = l3.next;
        }
        return dummy.next;
    }
}

File: Set1_1000/2_Add_Two_Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        dummyNode = ListNode(0)
        l3 = dummyNode
        carry = 0
        
        while(l1 != None or l2 != None):
            if l1 != None:
                l1Value = l1.val
            else:
                l1Value = 0
            
            if l2 != None:
                l2Value = l2.val
            else:
                l2Value = 0
                
            currentSum = l1Value + l2Value + carry
            carry = currentSum // 10
            lastDigit = currentSum % 10
            
            newNode = ListNode(lastDigit)
            l3.next = newNode
            
            if l1 != None:
                l1 = l1.next
            if l2 != None:
                l2 = l2.next
            l3 = l3.next
                
        if carry > 0:
            newPlaceDigit = ListNode(carry)
            l3.next = newPlaceDigit
            l3 = l3.next
            
        return dummyNode.next
                
            
                
